scripts/west_commands/runners/nrfjprog.py

```diff
@@ -150,58 +151,124 @@ def get_board_snr(self, glob):
 
+    def family_init(self):
+        # Set self.coprocessor according to self.family and
+        # self.build_conf, along with any other family specific
+        # initialization.
+
+        if self.family is None:
+            if self.build_conf.get('CONFIG_SOC_SERIES_NRF51X', False):
+                self.family = 'NRF51'
+            elif self.build_conf.get('CONFIG_SOC_SERIES_NRF52X', False):
+                self.family = 'NRF52'
+            elif self.build_conf.get('CONFIG_SOC_SERIES_NRF53X', False):
+                self.family = 'NRF53'
+            elif self.build_conf.get('CONFIG_SOC_SERIES_NRF91X', False):
+                self.family = 'NRF91'
+            else:
+                raise RuntimeError(f'unknown nRF; update {__file__}')
+
+        if self.family == 'NRF53':
+            if self.build_conf.get('CONFIG_SOC_NRF5340_CPUAPP', False):
+                self.coprocessor = 'CP_APPLICATION'
+            elif self.build_conf.get('CONFIG_SOC_NRF5340_CPUNET', False):
+                self.coprocessor = 'CP_NETWORK'
+            else:
+                raise RuntimeError(f'unknown nRF53; update {__file__}')
+            self.coprocessor_args = ['--coprocessor', self.coprocessor]
+        else:
+            self.coprocessor = None
+            self.coprocessor_args = []
+
+    def check_force_uicr(self):
+        # On SoCs without --sectoranduicrerase, we want to fail by
+        # default if the application contains UICR data and we're not sure
+        # that the flash will succeed.
+
+        # A map from SoCs which need this check to their UICR address
+        # ranges. If self.family isn't in here, do nothing.
+        uicr_ranges = {
+            'NRF53': ((0x00FF8000, 0x00FF8800),
+                      (0x01FF8000, 0x01FF8800)),
+            'NRF91': ((0x00FF8000, 0x00FF8800),),
+        }
+
+        if self.family not in uicr_ranges:
+            return
+
+        uicr = uicr_ranges[self.family]
+
+        if not self.force and has_region(uicr, self.hex_):
+            # Hex file has UICR contents.
+            raise RuntimeError(
+                'The hex file contains data placed in the UICR, which '
+                'needs a full erase before reprogramming. Run west '
+                'flash again with --force or --erase.')
+
+    def program_cmd(self, board_snr):
+        # Get the nrfjprog command use to actually program self.hex_.
+
+        # What type of erase argument should we pass to nrfjprog?
+        # The keys and values are:
+        # (family, coprocessor, --erase?): --nrfjprog-erase-argument
+        erase_args = {
+            ('NRF51', None, False): '--sectorerase',
+            ('NRF51', None, True): '--chiperase',
+
+            ('NRF52', None, False): '--sectoranduicrerase',
+            ('NRF52', None, True): '--chiperase',
+
+            # Mass erase with --recover is required to flash the
+            # network core using nrfjprog
+            ('NRF53', 'CP_NETWORK', True): '--recover',
```

Comment:
Recover should only be needed for the initial programming, after that the device will remain open. So it may still be usefull to leave the sectorerase option in place.

Reply:
Maybe we should then have a `--recover` option for the nrfjprog runner that would do this? So that it is not required to do the mass erase when only the network core is to be erased.

Reply:
> Maybe we should then have a `--recover` option for the nrfjprog runner that would do this?

Done.